lib/rubocop/cop/lint/void.rb

```diff
@@ -124,7 +124,8 @@ def check_void_expression(node)
         end
 
         def check_nonmutating(node)
-          return unless node.send_type? && NONMUTATING_METHODS.include?(node.method_name)
+          return unless (node.send_type? || node.block_type? || node.numblock_type?) &&
```

Comment:
Could we de-morgan this logic a bit maybe like so
```
wrong_type = !(node.send_type? || node.block_type? || node.numblock_type?)
wrong_method = !NONMUTATING_METHODS.include?(node.method_name)

return if wrong_type
return if wrong_method
```
or with the `unless` and the negation before the returns, but in two steps, I mean